### PR TITLE
[TACHYON-1048] Add Unit Tests for Async Eviction

### PR DESCRIPTION
**Pull request by shimingfei**
https://tachyon.atlassian.net/browse/TACHYON-1048

### CODE REVIEW

File: servers/src/test/java/tachyon/worker/block/TieredBlockStoreTestUtils.java [hunk 1]
```diff
@@ -267,6 +267,20 @@ public static void cache(long sessionId, long blockId, long bytes, StorageDir di
     }
   }
 
+  public static void cache(long sessionId, long blockId, long bytes, BlockStore blockStore,
```
Comment (calvinjia) [inline servers/src/test/java/tachyon/worker/block/TieredBlockStoreTestUtils.java]: Could you add a java doc for this utils method?

File: servers/src/test/java/tachyon/worker/block/SpaceReserverTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  private static final int[] TIER_LEVEL = {0, 1};
+  private static final StorageLevelAlias[] TIER_ALIAS = {StorageLevelAlias.MEM,
+      StorageLevelAlias.HDD};
+  private static final String[][] TIER_PATH = {{"/ramdisk"}, {"/disk1"}};
+  private static final long[][] TIER_CAPACITY_BYTES = {{1000}, {3000}};
+  BlockDataManager mBlockDataManager;
+  BlockStore mBlockStore;
+  SpaceReserver mSpaceReserver;
+  ExecutorService mExecutorService =
+      Executors.newFixedThreadPool(1, ThreadFactoryUtils.build("space-reserver-%d", true));
```
Comment (calvinjia) [inline servers/src/test/java/tachyon/worker/block/SpaceReserverTest.java]: Any reason why the last few variables are default visibility?

File: servers/src/test/java/tachyon/worker/block/SpaceReserverTest.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+      TieredBlockStoreTestUtils.cache(SESSION_ID, blockId ++, BLOCK_SIZE, mBlockStore,
+          tier0);
+    }
+    CommonUtils.sleepMs(WorkerContext.getConf()
+        .getLong(Constants.WORKER_SPACE_RESERVER_INTERVAL_MS));
+    BlockStoreMeta storeMeta = mBlockStore.getBlockStoreMeta();
+    Assert.assertEquals(3 * BLOCK_SIZE, storeMeta.getUsedBytes());
+    List<Long> usedBytesOnTiers = storeMeta.getUsedBytesOnTiers();
+    Assert.assertEquals(2 * BLOCK_SIZE,
+        (long)usedBytesOnTiers.get(StorageLevelAlias.MEM.getValue() - 1));
```
Comment (calvinjia) [inline servers/src/test/java/tachyon/worker/block/SpaceReserverTest.java]: minor style, space between cast and variable.

**Comment by shimingfei** [comment]
@calvinjia I have modified the code according to your comments, Thanks!

**Comment by shimingfei** [comment]
Jenkins, test this please

File: servers/src/test/java/tachyon/worker/block/SpaceReserverTest.java [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+  private static final long BLOCK_SIZE = 300;
+
+  private static final int[] TIER_LEVEL = {0, 1};
+  private static final StorageLevelAlias[] TIER_ALIAS = {StorageLevelAlias.MEM,
+      StorageLevelAlias.HDD};
+  private static final String[][] TIER_PATH = {{"/ramdisk"}, {"/disk1"}};
+  private static final long[][] TIER_CAPACITY_BYTES = {{1000}, {3000}};
+  BlockStore mBlockStore;
+  SpaceReserver mSpaceReserver;
+  ExecutorService mExecutorService = Executors.newFixedThreadPool(1);
```
Comment (calvinjia) [inline servers/src/test/java/tachyon/worker/block/SpaceReserverTest.java]: Any reason why these variables are default visibility?

Comment (shimingfei) [inline servers/src/test/java/tachyon/worker/block/SpaceReserverTest.java]: They should be private, Thanks!

**Comment by shimingfei** [comment]
@calvinjia 
I have changed the visibility of the variables

**Comment by calvinjia** [comment]
LGTM

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**